### ISSUE TITLE
Day Data not created some days

### ISSUE THREAD

**Issue by SBFspot** [SBFspot/SBFspot#116]
I'm seeing some strange behavior in V3.07.  During the last 10 days, SBFspot has failed to upload data to Pvoutput.org 6 days.  Spot csv files are being created every day and there are entries in the spot database every day.  However, on those days I see no data I have the following:

No Log file
regular .csv file is missing or only contains 1-2 entries
Daydata table has no entries in DB

I'm not sure why this is happening, all the spot data is present in the .csv files and in the DB, so communication with the inverter is just fine.

Thanks,

Pete

#### Attachments
[catchout.v307-noday.txt](https://www.codeplex.com/Download/AttachmentDownload.ashx?ProjectName=SBFspot&WorkItemId=111&FileAttachmentId=1458586)
[catchout.v307-10day.txt](https://www.codeplex.com/Download/AttachmentDownload.ashx?ProjectName=SBFspot&WorkItemId=111&FileAttachmentId=1458587)
[Catch Output.zip](https://www.codeplex.com/Download/AttachmentDownload.ashx?ProjectName=SBFspot&WorkItemId=111&FileAttachmentId=1458613)
[TEST-C111.zip](https://www.codeplex.com/Download/AttachmentDownload.ashx?ProjectName=SBFspot&WorkItemId=111&FileAttachmentId=1458731)
[BartPlant.zip](https://www.codeplex.com/Download/AttachmentDownload.ashx?ProjectName=SBFspot&WorkItemId=111&FileAttachmentId=1459424)
[catchout.6-5-15-days.txt](https://www.codeplex.com/Download/AttachmentDownload.ashx?ProjectName=SBFspot&WorkItemId=111&FileAttachmentId=1460468)
[Bartplant.zip](https://www.codeplex.com/Download/AttachmentDownload.ashx?ProjectName=SBFspot&WorkItemId=111&FileAttachmentId=1463866)
[capt-31-21-ou.txt](https://www.codeplex.com/Download/AttachmentDownload.ashx?ProjectName=SBFspot&WorkItemId=111&FileAttachmentId=1463996)

#### This work item was migrated from CodePlex
CodePlex work item ID: '111'
Vote count: '1'

**Comment by SBFspot** [comment]
[SillieWimons@23/05/2015]
dear pete,        near certain you're also confronted with issue c108

crash  for  -adnn  and  default

v308 is now in test   and migth be released soon   (v308 solves that problem)

for the time being you have to step back to version 305 or wait till 308 is released

kr wim

**Comment by SBFspot** [comment]
[pbartko@23/05/2015]
Just to add, I'm seeing this behavior on Raspi.  

I'm also willing to help with Beta testing, just let me know.

Thanks,

Pete

**Comment by SBFspot** [comment]
[SillieWimons@24/05/2015]
dear pete,       it is expected that issue c108  is related with type of inverter  (not with OperSyst)

to find out whether youre problem is == c108   we like to obtain the following information

-a- your working envirnment:  -type inverter, -operating system; -PC  ==RPi
-b- traces (stdout and stderr file)  for the following commands  (cfr the version of sbfspot)
xxxxx stands for the options in your own setup,  
v305  stands for any version prior to v306  that your were using  (even v24x is ok)
allow csv output and include the csv files (the one that you extract with v305 and v308)
these files do make it easy to interprete the details in the packets that are received form the inverter

-b.1-  sbfspot.308 -d5 -v5 -ad0 -am1  xxxxxx  1>catchout.v308-noday.txt   2>catcherr.v308-noday.txt
-b.2-  sbfspot.308 -d5 -v5 -sp0 -ad10 -am0  xxxxxx  1>catchout.v308-10day.txt   2>catcherr.v308-10day.txt  
-b.3-  sbfspot.305 -d5 -v5 -ad10 -am1  xxxxxx  1>catchout.v305-10day.txt   2>catcherr.v305-10day.txt

**Comment by SBFspot** [comment]
[pbartko@24/05/2015]
a) Inverter is SMA SB3000US model with the SMA bluetooth PiggyBack module installed.  Platform is RPI. 

b) I ran the following commands and captured the output:

SBFspot -d5 -v5 -ad0 -am1 1>catchout.v307-noday.txt 2>catcherr.v307-noday.txt

SBFspot -d5 -v5 -sp0 -ad10 -am0 1>catchout.v307-10day.txt 2>catcherr.v307-10day.txt

Error files are empty, Output files are attached.

I will have to recompile V3.05 in order to get its output.

Thanks,

Pete

**Comment by SBFspot** [comment]
[pbartko@24/05/2015]
10 day file uploaded

**Comment by SBFspot** [comment]
[SillieWimons@24/05/2015]
dear pete,        from the information that you did send     it is 98% certain the C108 issue

but ??? what are the tricks that some days do get through   ???

-a- inverters with piggy back are 99,9% candidate to be confronted with the issue c108 (== yours),
-b- your stdout   is cut off at the same point where it went wrong with my inverter as well,
!!! when sbfspot hangs then the "text" part in treatment for the stdout is not yet "flushed" to the file
for that reason the "moment" when the problem occurs is not yet available in the file
the only way to capture the complete output is: -execute the task from a "command-box"
apply the command without file redirect:      SBFspot -d5 -v5 -sp0 -ad10 -am0 -ae2

-c- to allow us some further / additional understanding   ????
rational:
-c.1- in your system - some of the days succeed to get interrogated  :-)
in my system i did have a 100% failure rate     ;-{
-c.2- can you execute the following command   (allow .csv output) 
SBFspot -d5 -v5 -ad0 -ae2 1>catchout.v307-archevent.txt 2>catcherr.v307-archevent.txt
and send the stdout and the user event .csv files

kr wim

**Comment by SBFspot** [comment]
[SBF@24/05/2015]
Beta 3.0.8 has been released today - this version should solve your problem

**Comment by SBFspot** [comment]
[pbartko@24/05/2015]
I ran the command:

SBFspot -d5 -v5 -ad0 -ae2 1>catchout.v307-archevent.txt 2>catcherr.v307-archevent.txt

and have uploaded the output files, including the csv

I'll download V3.08 and give it a try.

Thanks,

Pete

**Comment by SBFspot** [comment]
[SillieWimons@25/05/2015]
dear pete,       done further analysis on your results       !!!!

-a- ANOTHER (new) issue in your system   -> NO MONTH files
please look into the "noday" output and the "archevents" output   text states
" ExportMonthDataToCSV()      There is no data to export! "

-b- can you inform about the following,        I added all questions that might deliver usefull info
-b.1- is there another tool active in your system  (SunnyBeam,  SunExplorer, . . . )
-b.2- since when did you have the problem for daily files   (only since v307???)
-b.3- when did you start using v307
-b.4- did you ever before obtain month files from you inverter    format name:   BartPlant-yyyymm.csv
if so when did it stop  (eg till march2015)
was the last file a complete one   (till end of month) 

-c- after installing v308    can you execute the below commands
-c.1-  can you execute the below commands    with your actual config file
SBFspot -d5 -v5 -am0 -ad6 -ae0 -nosql 1>catchout.v308-days.txt 2>catcherr.v308-days.txt
this command should inform whether all daily info is ok (by including the days that did show errors)
-c.2- create a second cfg file (in same directory as the exec) - apply a seperate directory for .csv files
OutputPath=/home/pi/smadata/TEST-C111
OutputPathEvents=/home/pi/smadata/TEST-C111/Events
execute the below command  (syntax see help    -?  )
SBFspot -d5 -v5 -am3 -ad10 -ae3 -noSQL -cfgFULLFILENAME 1>catchout.v308-day-event.txt 2>catcherr.v308-day-event.txt
SBFspot -d5 -v5 -am0 -ad0 -ae3 -noSQL -cfgFULLFILENAME -installer -password:1111 1>catchout.v308-instal.txt 2>catcherr.v308-instal-event.txt
rational: we don't yet know what is wrong -> do not touch older .csv files  NOR sql-db
please sent output files and csv files:
- user and installer events are written to .csv files in the directory TEST-C111/events
note I do assume standard password setting for installer

kr wim

**Comment by SBFspot** [comment]
[pbartko@25/05/2015]
b.1. Yes, I have a SunnyBeam recording data from the inverter also.
b.2. Daily file issue started with V3.07
b.3. I installed V3.07 as soon as it came out because I was having issues with high power output values being rejected by pvoutput.org.
b.4. No, I do not have any month files from a year of using the software.  I began using using SMAspot last year, up until April 3, 2015 when I installed SBFspot V3.03.  Upgraded to V3.07 on April 28 and now am running V3.08.

c.2. I ran both commands using V3.08 and have uploaded zip file with results

Thanks,

Pete

**Comment by SBFspot** [comment]
[SillieWimons@26/05/2015]
dear pete,     allow me to raise some more questions    - read about "echternach's dancing procession" :-)

from your information I do assume that:
-a.1- your system does execute its SBFspot tasks without crash
-a.2- upload to PVoutput might  run correctly    (database seems correct)
note it might be that you need to execute once SBFspot using the option -adnn
to complete the database for the missing nn day
-a.3- there are "strange" effects in your system, for that reason 

after checking your info , I did make tests with v308 (25-05-2015) and compare results to v305,
rational: in your stdout, a lot of "twins" records and "missingdaily" records  are mentioned,
i did want to check (within my system) that SBFspot v308 did not mask info

kr wim   (added inf below)

-b- on my system - the "twins" records - are treated correct by SBFspot
BUT i do have only ONCE  a day  the message for "twins" == "overwriting" record ???
-a.1- I'll check on my system what the trigger for "twins" might be

-b- in my system i do also use a SunnyBeam (SB-11 BlueTooth)
but SMA has shipped several variants of SunBeam
-b.1- SunnyBeam is handy as tool - because when out of home it keeps my inverter clock up-to-date
-b.2- ?? which type / version / Firmware-version is part of your system
-b.3- ?? on my SunnyBeam-11  - my SB11 stores a copy of the monthly and daily-10m files
note about daily-10m ->my SB11 stores the daily file with 10minutes interval
what about your's ???

-c- in your system there are several "missingdaily"   AND   several "Twins"  in one day
-c.1- i'm now point/point checking the outputs
this will take time before complement remarks

**Comment by SBFspot** [comment]
[pbartko@28/05/2015]
a.
SBFspot V3.08 is now running without a crash.  I did use -adnn option to backload all the missing data from the inverter.

b.
Sunny Beam Firmware 01.07.06.R
Hardware: A1
Bluetooth Version: V1.208.4
Mine Stores 10 minute data, and also produces monthly files.

c. I notices the missing and twins data also.

Thanks,

Pete

**Comment by SBFspot** [comment]
[SillieWimons@29/05/2015]
hello pete    in summary

-a- SBFspot v308   fits to your needs

-b- remaining point / issue to investigate ->  why does SBFspot not produce monthly files at all ???
cfr messages   " ExportMonthDataToCSV() There is no data to export! "
-b.1-you do have an escape via the "SunnyBeam" files

-c- for investigation -> ?? is it abnormal  or  correct   that in your system there are 
-c.1- more then expected "missings"  
-c.2- more then expected "overwrties"  (alias twins)

-d- for the record,   can you make use of SunnyExplorer ??
this in case, we might need a check on info that is not (yet) supported by SBFspot

kr wim

**Comment by SBFspot** [comment]
[pbartko@30/05/2015]
Currently SBFspot v3.08 is working well and suits my needs

I don't know why I'm not getting monthly files.  Its not important as you say, I do have the Sunny Beam monthly files.

I'm not sure why I have missing and twin data points.  

I do have Sunny Explorer installed on my PC and have used it to diagnose previous problems.

Thanks,

Pete

**Comment by SBFspot** [comment]
[SillieWimons@31/05/2015]
dear pete,      after checking through the texts - allow me to

** can you confirm for the values as seen by sunny explorer
?? does sunny explorer show  monthly files  ??
?? can you provide the daily files   as  retrieved by sunny explorer   eg the ones 16may to 25may

** can you (slightly) modify the cron tasks  for sbfspot and upload  (check syntax !!!  :-)
rational: the internal organisation of the inverter does use a 5minutes cycle
at end of the cycle the values for "daily" are written
the proposed change makes sbfspot run at minute2 in 5min cycle  and  upload at the min3
2-57/5 * * * * sbfspot . . . 
3-58/5 * * * * upload . . . .
this inhibits possible interference between SBFspot and the internals of the inverter

** distinquish the issues as mentioned below:

kr wim

-a- issue C105  values high (caused by missing records)
comment-a.1- solution works fine if not combined with "twins-records"

-b- issue c108   solved in v308  "SBFspot hangs for -adnn"    nickname "twins-records"
comment-b.1- c108 was an unpredicted result from solving issue c105
comment-b.2- from c111 there are several variants in twins
-b.2-1- twins due to reset of time - cfr my inverter - same time (zero|low production) - no missing records
the solution -> drop the first record works correct
-b.2-2- twins during production - cfr pete-bartko - same time (high production) - mix with missing rec
the combined solution for c105 and c111 creates strange values

-c- issue c111  (alias c108)  this is solved in v308
from the analysis  it is found that there are subsequent issues

-d- variant for "twins records" during production,    
the second record seems to have the correct production  value but a wrong time
comment-d.1-    i recommend to check for the values with v308   <->  sunny explorer 
and if needed to open a new issue

-e- combination of "twins"  and  "missing
comment-e.1-  for a large number of examples it seems that 

-f- month files not created
i propose to handle this as a seperate issue

**Comment by SBFspot** [comment]
[pbartko@31/05/2015]
I checked using Sunny Explorer.  I downloaded the daily files for 5/16-5/25 and attached them in a zip file.  I also tried downloading monthly files, but it did not work.  Sunny Explorer had the following output:

Read monthly data from April
Read monthly data from March
Read monthly data from February
Read monthly data from January
Data export completed

But it failed to write the monthly files to the hard drive like it did for the daily files.  When downloading the daily files, the output was similar but after exporting the data it gave a message that it was writing each day to a csv file.  Those outputs are missing for the monthly files.

I'll modify the CRON tasks to see if it has any affect.

Thanks,

Pete

**Comment by SBFspot** [comment]
[SillieWimons@31/05/2015]
dear pete      thank you for your work

may i ask (give it sufficient time to make it certain) to confirm that the upload works correctly

please understand that:
- detailed knowledge about the data and protocol has been gathered via reverse engineering
- treatment of day-day operations, for most inverter types, has been implemented,

analysis of exception cases, as your's, add valuable complement information

kr wim

**Comment by SBFspot** [comment]
[pbartko@5/06/2015]
Here is an update,

Changing the CRON task times didn't have an effect.  I let the system run all week and today I pulled a log file (see "catchout.6-5-15-days.txt").  It still shows twin records and data holes in particular for today.  It ran for 3 days with no problems, then there was a lot of them today.

The command I used for the log file is:
 /usr/local/bin/sbfspot.3/SBFspot -d5 -v5 -am0 -ad6 -ae0 -nosql 1>catchout.6-5-15-days.txt 2>catcherr.6-5-15-days.txt

The Catcherr file was empty.

Thanks,
Pete

**Comment by SBFspot** [comment]
[SillieWimons@12/06/2015]
dear pete,   thank you for the log file,

if your cron is still running in the same mode   :-)   , can you
-a.1- change for once the task into   -ad14    to capture the full period from 31may till now
-a.2- after the above you can modify the task  and put a value for the option that is convenient for you
 
can you provide, for the further analysis and evaluation, 
-b- the daily .csv files for the period since 31may
-c- the spot .csv files for the period since 31may
-c.1- the spot .csv files (with a 2min shift your task captures spot values ~middle of the 5min cycle)
might render additional info when comparing with daily
-c.2- spot .csv files act as a "log" file for the execution of the SBFspot 
-d- the event .csv files - user and installer - (you'll need to execute this seperate) for may-june

kind regards    wim

**Comment by SBFspot** [comment]
[pbartko@20/06/2015]
wim,

I've uploaded all the files you requested in the latest .zip file.  I still periodically have holes and twins records but V3.08 handles them without any problems.  I have noticed that when I have those, Pvoutput does not show the Average trace on the graph correctly.  I assume that is a fault of pvoutput.org?

Thanks,

Pete

**Comment by SBFspot** [comment]
[pbartko@20/06/2015]
wim,

I've uploaded the output file you requested using the following command:

/usr/local/bin/sbfspot.3/SBFspot  -d5 -v5 -ad23  1>capt-31-21-ou.txt 2>capt-31-21-er.txt

I didn't include the error file because it is empty.

Pete

**Comment by SBFspot** [comment]
[SBF@21/06/2015]
> Pvoutput does not show the Average trace on the graph correctly. I assume that is a fault of pvoutput.org? 

Can you share the link?

**Comment by SBFspot** [comment]
[pbartko@22/06/2015]
SBF

Here is one of the days

http://pvoutput.org/intraday.jsp?id=1885&sid=1421&dt=20150605

It is strange that pvoutput only shows average plot when not logged into my account.  If I log in, it does not show the average data.

Thanks

Pete

**Comment by SBFspot** [comment]
[SBF@23/06/2015]
Indeed a strange graph...
If you like, you can ask some clarification at https://getsatisfaction.com/pvoutput
Bankstownbloke will have the answer, pretty sure
